Importing .ts files in app.config.ts crashes with ERR_MODULE_NOT_FOUND

### Minimal reproducible example

Reproducible example: https://github.com/flodaniel/expo-53-ts-bug


If `app.config.ts` imports `a.ts` which then imports `b.ts`, the import of `b.ts` is not found:

```
Error [ERR_MODULE_NOT_FOUND]: Error reading Expo config at /PATH_TO_PROJECT/app.config.ts:

Cannot find module '/PATH_TO_PROJECT/src/config/b.ts' imported from /PATH_TO_PROJECT/src/config/a.ts
```



### Steps to reproduce

1. Clone https://github.com/flodaniel/expo-53-ts-bug
2. install deps and run `npm run start`
3. observe crash caused by the import of b.ts in a.ts


### Environment

```text
expo-env-info 1.3.3 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 22.16.0 - ~/.volta/tools/image/node/22.16.0/bin/node
      Yarn: 1.22.22 - ~/.volta/tools/image/yarn/1.22.22/bin/yarn
      npm: 10.9.2 - ~/.volta/tools/image/node/22.16.0/bin/npm
      Watchman: 2024.12.02.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.16.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.2, iOS 18.2, macOS 15.2, tvOS 18.2, visionOS 2.2, watchOS 11.2
      Android SDK:
        API Levels: 28, 29, 30, 31, 32, 33, 33, 34, 35
        Build Tools: 29.0.2, 30.0.3, 33.0.2, 34.0.0, 35.0.0
        System Images: android-26 | Google APIs ARM 64 v8a, android-26 | Google APIs Intel x86 Atom, android-26 | Google APIs Intel x86_64 Atom, android-28 | Google APIs ARM 64 v8a, android-29 | Google APIs ARM 64 v8a, android-33 | Google APIs ARM 64 v8a, android-34 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2024.2 AI-242.23726.103.2422.12816248
      Xcode: 16.2/16C5032a - /usr/bin/xcodebuild
    npmPackages:
      expo: ~53.0.9 => 53.0.9 
      react: 19.0.0 => 19.0.0 
      react-native: 0.79.2 => 0.79.2 
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

```text
15/15 checks passed. No issues detected!

(only works when import of b.ts is removed in a.ts)
```

--- a/docs/pages/config-plugins/development-and-debugging.mdx
+++ b/docs/pages/config-plugins/development-and-debugging.mdx
@@ -513,7 +513,7 @@ The Expo CLI `npx expo prebuild` command uses [`@expo/prebuild-config`][prebuild
 
 For example, say you wanted to add support for managing the `ios/*/AppDelegate.h` file, you could do this by adding a `ios.appDelegateHeader` modifier.
 
-> This example uses `ts-node` for simple local TypeScript support, this isn't strictly necessary. [Learn more](/guides/typescript/#appconfigjs).
+> This example uses `tsx` for simple local TypeScript support, this isn't strictly necessary. [Learn more](/guides/typescript/#appconfigjs).
 
 ```ts withAppDelegateHeaderBaseMod.ts
 import { ConfigPlugin, IOSConfig, Mod, withMod, BaseMods } from 'expo/config-plugins';
@@ -578,7 +578,7 @@ To use this new base mod, add it to the plugins array. The base mod **MUST** be 
 
 ```js app.config.js
 // Required for external files using TS
-require('ts-node/register');
+require('tsx/cjs');
 
 import {
   withAppDelegateHeaderBaseMod,

--- a/docs/pages/config-plugins/plugins.mdx
+++ b/docs/pages/config-plugins/plugins.mdx
@@ -135,14 +135,14 @@ export default withPlugin;
 
 We recommend writing config plugins in TypeScript, since this will provide intellisense for the configuration objects. However, your app config is ultimately evaluated by Node.js, which does not recognize TypeScript code by default. Therefore, you will need to add a parser for the TypeScript files from the the **plugins** directory to **app.config.ts** file.
 
-Install `ts-node` library by running the following command:
+Install `tsx` library by running the following command:
 
-<Terminal cmd={['$ npm install --save-dev ts-node']} />
+<Terminal cmd={['$ npm install --save-dev tsx']} />
 
 Then, change the static app config (**app.json**) to the [dynamic app config (**app.config.ts**)](/workflow/configuration/#dynamic-configuration) file. You can do this by renaming the **app.json** file to **app.config.ts** and changing the content of the file as shown below. Ensure to add the following import statement at the top of your **app.config.ts** file:
 
 ```ts app.config.ts
-import 'ts-node/register';
+import 'tsx/cjs';
 
 module.exports = () => {
   /* @hide ... rest of your app config */
@@ -159,7 +159,7 @@ module.exports = () => {
 Now, you can call the config plugin from your dynamic app config. To do this, you need to add the path to the **withPlugin.ts** file to the plugins array in your app config:
 
 ```ts app.config.ts
-import "ts-node/register";
+import "tsx/cjs";
 /* @info */
 import { ExpoConfig } from "expo/config";
 /* @end */

--- a/docs/pages/guides/typescript.mdx
+++ b/docs/pages/guides/typescript.mdx
@@ -185,19 +185,19 @@ Some Expo libraries provide both static types and type generation capabilities. 
 
 ## TypeScript for project's config files
 
-Additional setup is required to use TypeScript for configuration files such as **metro.config.js** or **app.config.js**. You need to utilize [`ts-node` require hook](https://github.com/TypeStrong/ts-node#programmatic) to import TypeScript files within your JS config file. This hook allows TypeScript imports while keeping the root file as JavaScript.
+Additional setup is required to use TypeScript for configuration files such as **metro.config.js** or **app.config.js**. You need to utilize [`tsx/cjs` require hook](https://tsx.is/dev-api/entry-point#commonjs-mode-only) to import TypeScript files within your JS config file. This hook allows TypeScript imports while keeping the root file as JavaScript.
 
 <Tabs>
 
 <Tab label="macOS/Linux">
 
-<Terminal cmd={['$ npx expo install ts-node --dev']} />
+<Terminal cmd={['$ npx expo install tsx --dev']} />
 
 </Tab>
 
 <Tab label="Windows">
 
-<Terminal cmd={['$ npx expo install ts-node "--" --dev']} />
+<Terminal cmd={['$ npx expo install tsx "--" --dev']} />
 
 </Tab>
 
@@ -208,7 +208,7 @@ Additional setup is required to use TypeScript for configuration files such as *
 Update **metro.config.js** to require **metro.config.ts** file:
 
 ```js metro.config.js
-require('ts-node/register');
+require('tsx/cjs');
 module.exports = require('./metro.config.ts');
 ```
 
@@ -227,7 +227,7 @@ module.exports = config;
 Install the `@expo/webpack-config` package.
 
 ```js webpack.config.js
-require('ts-node/register');
+require('tsx/cjs');
 module.exports = require('./webpack.config.ts');
 ```
 
@@ -249,7 +249,7 @@ module.exports = async function (env: Environment, argv: Arguments) {
 **app.config.ts** is supported by default. However, it doesn't support external TypeScript modules, or **tsconfig.json** customization. You can use the following approach to get a more comprehensive TypeScript setup:
 
 ```ts app.config.ts
-import 'ts-node/register'; // Add this to import TypeScript files
+import 'tsx/cjs'; // Add this to import TypeScript files
 import { ExpoConfig } from 'expo/config';
 
 const config: ExpoConfig = {

--- a/docs/pages/workflow/configuration.mdx
+++ b/docs/pages/workflow/configuration.mdx
@@ -69,7 +69,7 @@ Library authors can extend the app config by using [Expo Config plugins](/config
 For more customization, you can use the JavaScript (**app.config.js**) or [TypeScript](#using-typescript-for-configuration-appconfigts-instead-of-appconfigjs) (**app.config.ts**). These configs have the following properties:
 
 - Comments, variables, and single quotes.
-- ESM import syntax (the `import` keyword) is not supported, except when using [TypeScript with `ts-node`](/guides/typescript/#appconfigjs). JS files that are compatible with your current version of Node.js can be imported with `require()`.
+- ESM import syntax (the `import` keyword) is not supported, except when using [TypeScript with `tsx`](/guides/typescript/#appconfigjs). JS files that are compatible with your current version of Node.js can be imported with `require()`.
 - TypeScript support with nullish coalescing and optional chaining.
 - Updated whenever Metro bundler reloads.
 - Provide environment information to your app.
@@ -161,7 +161,7 @@ export default ({ config }: ConfigContext): ExpoConfig => ({
 });
 ```
 
-To import other TypeScript files into **app.config.ts** or customize the language features, we recommend using [`ts-node`](/guides/typescript/#appconfigjs). `ts-node` also enables using `import` syntax in any file imported by **app.config.ts**. This means you can write local [config plugins](/config-plugins/introduction/) in TypeScript with full language features.
+To import other TypeScript files into **app.config.ts** or customize the language features, we recommend using [`tsx`](/guides/typescript/#appconfigjs). `tsx` also enables using `import` syntax in any file imported by **app.config.ts**. This means you can write local [config plugins](/config-plugins/introduction/) in TypeScript with full language features.
 
 ### Configuration resolution rules
 

